File: src/main/java/am9/olbcore/Account.java
package am9.olbcore;

import java.sql.ResultSet;
import java.sql.SQLException;
import java.sql.Statement;
import am9.olbcore.UniversalLogger;

public class Account {
    public static void create(String name, Integer uid) {
        try {
            Statement s = am9.olbcore.Database.c.createStatement();
            s.execute("insert into user (name, id, banned) VALUES (" + name + ", " + uid + ", 0)");
        } catch (Exception e) {
            if (am9.olbcore.Database.c == null) {
                throw new NullPointerException("未连接数据库");
            } else {
                UniversalLogger.Companion.error(e.getMessage());
            }
        }
    }
    public static void ban(Integer uid) {
        try {
            Statement s = am9.olbcore.Database.c.createStatement();
            s.execute("update user set banned = 1 where id = " + uid);
        } catch (Exception e) {
            if (am9.olbcore.Database.c == null) {
                throw new NullPointerException("未连接数据库");
            } else {
                UniversalLogger.Companion.error(e.getMessage());
            }
        }
    }
    public static void pardon(Integer uid) {
        try {
            Statement s = am9.olbcore.Database.c.createStatement();
            s.execute("update user set banned = 0 where id = " + uid);
        } catch (Exception e) {
            if (am9.olbcore.Database.c == null) {
                throw new NullPointerException("未连接数据库");
            } else {
                UniversalLogger.Companion.error(e.getMessage());
            }
        }
    }
    public static void delete(Integer uid) {
        try {
            Statement s = am9.olbcore.Database.c.createStatement();
            s.execute("delete from user where id = " + uid);
        } catch (Exception e) {
            if (am9.olbcore.Database.c == null) {
                throw new NullPointerException("未连接数据库");
            } else {
                UniversalLogger.Companion.error(e.getMessage());
            }
        }
    }
    public static void update(Integer uid, String name) {
        try {
            Statement s = am9.olbcore.Database.c.createStatement();
            s.execute("update user set name = " + name + " where id = " + uid);
        } catch (Exception e) {
            if (am9.olbcore.Database.c == null) {
                throw new NullPointerException("未连接数据库");
            } else {
                UniversalLogger.Companion.error(e.getMessage());

            }
        }
    }
    public static String query(Integer uid) {
        try {
            Statement s = am9.olbcore.Database.c.createStatement();
            s.execute("select name from user where id = " + uid);
            ResultSet result = s.getResultSet();
            if (result.next()) {
                return result.getString(1);
            } else {
                return "nothing";
            }
        } catch (Exception e) {
            if (am9.olbcore.Database.c == null) {
                throw new NullPointerException("未连接数据库");
            } else {
                UniversalLogger.Companion.error(e.getMessage());
            }
            return "nothing";
        }
    }
    public static Integer query(String name) {
        try {
            Statement s = am9.olbcore.Database.c.createStatement();
            s.execute("select id from user where name = " + name);
            ResultSet result = s.getResultSet();
            if (result.next()) {
                return result.getInt(1);
            } else {
                return -1;
            }
        } catch (SQLException e) {
            return -1;
        }
    }
}

File: src/main/java/am9/olbcore/Account.kt
package am9.olbcore

import java.sql.*
import am9.olbcore.UniversalLogger

class Account {
    companion object {
        @Throws(SQLException::class, NullPointerException::class)
        @JvmStatic
        fun create(name: String, uid: Int) {
            val s = am9.olbcore.Database.c.createStatement()
            s.executeUpdate("INSERT INTO user (name, id, banned, qid, mcname) VALUES ($name, $uid, 0, null, null)")
        }

        @Throws(SQLException::class, NullPointerException::class)
        @JvmStatic
        fun createWithQQ(name: String, uid: Int, qid: Long) {
            val s = am9.olbcore.Database.c.createStatement()
            s.executeUpdate("INSERT INTO user (name, id, banned, qid, mcname) VALUES ($name, $uid, 0, $qid, null)")
        }

        @Throws(SQLException::class, NullPointerException::class)
        @JvmStatic
        fun createWithMC(name: String, uid: Int, mcname: String) {
            val s = am9.olbcore.Database.c.createStatement()
            s.executeUpdate("INSERT INTO user (name, id, banned, qid, mcname) VALUES ($name, $uid, 0, null, $mcname)")
        }

        @Throws(SQLException::class, NullPointerException::class)
        @JvmStatic
        fun ban(uid: Int) {
            val s = am9.olbcore.Database.c.createStatement()
            s.executeUpdate("UPDATE user SET banned = 1 WHERE id = $uid")
        }

        @Throws(SQLException::class, NullPointerException::class)
        @JvmStatic
        fun pardon(uid: Int) {
            val s = am9.olbcore.Database.c.createStatement()
            s.executeUpdate("UPDATE user SET banned = 0 WHERE id = $uid")
        }

        @Throws(SQLException::class, NullPointerException::class)
        @JvmStatic
        fun delete(uid: Int) {
            val s = am9.olbcore.Database.c.createStatement()
            s.executeUpdate("DELETE FROM user WHERE id = $uid")
        }

        @Throws(SQLException::class, NullPointerException::class)
        @JvmStatic
        fun update(uid: Int, name: String) {
            val s = am9.olbcore.Database.c.createStatement()
            s.executeUpdate("UPDATE user SET name = $name WHERE id = $uid")
        }

        @Throws(SQLException::class, NullPointerException::class)
        @JvmStatic
        fun query(uid: Int): String {
            val s = am9.olbcore.Database.c.createStatement()
            s.execute("SELECT name FROM user WHERE id = $uid")
            val result = s.executeQuery()
            return if (result.next()) result.getString(1) else "nothing"
        }

        @Throws(SQLException::class)
        @JvmStatic
        fun queryByName(name: String): Int {
            val s = am9.olbcore.Database.c.createStatement()
            s.execute("SELECT id FROM user WHERE name = $name")
            val result = s.executeQuery()
            return if (result.next()) result.getInt(1) else -1
        }

        @Throws(SQLException::class, NullPointerException::class)
        @JvmStatic
        fun bindQQ(uid: Int, qid: Long) {
            val s = am9.olbcore.Database.c.createStatement()
            s.executeUpdate("UPDATE user SET qid = $qid WHERE id = $uid")
        }

        @Throws(SQLException::class, NullPointerException::class)
        @JvmStatic
        fun bindMC(uid: Int, mcname: String) {
            val s = am9.olbcore.Database.c.createStatement()
            s.executeUpdate("UPDATE user SET mcname = $mcname WHERE id = $uid")
        }
    }
}
